demos/multi_state_bot/demo.py

```diff
@@ -0,0 +1,91 @@
+"""The module is a script to run the bot."""
+
+from hammett.core import Application, Button, Screen
+from hammett.core.constants import DEFAULT_STATE, RenderConfig, SourcesTypes
+from hammett.core.handlers import register_typing_handler
+from hammett.core.mixins import RouteMixin, StartMixin
+from hammett.core.persistence import RedisPersistence
+
+ANONYMOUS_SCREEN_DESCRIPTION = (
+    'Hello, <b>Anonymous</b>!\n'
+    '\n'
+    '<i>The bot is in the <b>DEFAULT</b> state now</i>.'
+)
+
+INTRODUCTION_SCREEN_WITH_NAME_DESCRIPTION = (
+    'Hi, <b>{name}</b>!\n'
+    '\n'
+    '<i>The bot has switched back to the <b>DEFAULT</b> state</i>.'
+)
+
+INTRODUCTION_SCREEN_WITHOUT_NAME_DESCRIPTION = (
+    "What's your name?\n"
+    "\n"
+    "<i>The bot is now switched to the <b>TYPE_NAME</b> state</i>."
+)
+
+TYPE_NAME_STATE = 'type_name_state'
+
+
+class AnonymousScreen(StartMixin):
+    """The class implements AnonymousScreen. It will be registered
+    in the DEFAULT_STATE.
+    """
+
+    description = ANONYMOUS_SCREEN_DESCRIPTION
+
+    async def add_default_keyboard(self, _update, _context):
+        """Set up the keyboard for the screen."""
+        return [[
+            Button(
+                'Introduce Yourself',
+                IntroductionScreen,
+                source_type=SourcesTypes.MOVE_ALONG_ROUTE_SOURCE_TYPE,
+            ),
+        ]]
+
+
+class IntroductionScreen(RouteMixin, Screen):
+    """The class implements IntroductionScreen. It will be registered
+    in the TYPE_NAME_STATE.
+    """
```

Comment:
`"""The class implements IntroductionScreen. It will be registered in TYPE_NAME_STATE."""`